src/NUnitFramework/framework/Constraints/Numerics.cs

```diff
@@ -418,5 +418,75 @@ public static int Compare(object expected, object actual)
             return Convert.ToInt32(expected).CompareTo(Convert.ToInt32(actual));
         }
         #endregion
+
+        #region Numeric Difference
+
+        /// <summary>
+        /// Calculates the difference between 2 values.
+        /// </summary>
+        /// <param name="expected">The expected value</param>
+        /// <param name="actual">The actual value</param>
+        /// <param name="tolerance">A reference to the tolerance in effect</param>
+        /// <returns>The difference between the values</returns>
+        public static object Difference(object expected, object actual, Tolerance tolerance)
+        {
+            switch (tolerance.Mode)
+            {
+                case ToleranceMode.Linear:
+                    return DifferenceAbsolute(expected, actual);
+                case ToleranceMode.Percent:
+                    return DifferencePercent(expected, actual);
+                default:
+                    throw new InvalidOperationException("Cannot calculate a difference for specified tolerance mode");
+            }
+        }
+
+        private static object DifferencePercent(object expected, object actual)
```

Comment:
I was comparing this method to the method below, and I think it makes more sense just to have one method that handles both cases (rather than two the have more or less the same flow). This method could be passed e.g. a boolean to determine if the result should be absolute or in percent. Then the case for decimal would become:
```csharp
...
            if (expected is decimal || actual is decimal)
            {
                var difference = Convert.ToDecimal(expected) - Convert.ToDecimal(actual);
                return isAbsolute ? difference : difference / Convert.ToDecimal(expected) * 100;
            }
...
```

Reply:
Methods merged, looks bulky a bit but at the same time more convenient. I think the best way is to rework all this stuff using generics.